### PR TITLE
example: add semantic cache demo built with jinaai and tidb vector

### PR DESCRIPTION
**Pull request by sykp241095**

### CODE REVIEW

File: examples/semantic-cache/cache.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    key_vec = generate_embeddings(credentials.credentials, key)
+    # The max value of distance is 0.3
+    max_distance = min(max_distance, 0.3)
+
+    with Session(engine) as session:
+        result = session.exec(
+            select(
+                Cache,
+                Cache.key_vec.cosine_distance(key_vec).label('distance')
+            ).filter(
```
Comment (wd0517) [inline examples/semantic-cache/cache.py:127]: The ANN index will not be enabled when using a pre-filter.

File: examples/semantic-cache/cache.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+from datetime import datetime
+from typing import Optional, Annotated
+
+import requests
+import dotenv
+from fastapi import Depends, FastAPI
+from fastapi.security import HTTPAuthorizationCredentials, HTTPBearer
+from sqlmodel import (
+    SQLModel,
+    Session,
```
Comment (wd0517) [inline examples/semantic-cache/cache.py:11]: Please use `AsyncSession`

File: examples/semantic-cache/cache.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        )
+    )
+
+engine = create_engine(DATABASE_URI)
+SQLModel.metadata.create_all(engine)
+
+app = FastAPI()
+security = HTTPBearer()
+
+readme = open("README.md").read()
```
Comment (wd0517) [inline examples/semantic-cache/cache.py:79]: not used

File: examples/semantic-cache/cache.py [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        'input': [text],
+        'model': 'jina-embeddings-v2-base-en'  # with dimisions 768
+    }
+    response = requests.post(JINAAI_API_URL, headers=JINAAI_HEADERS, json=JINAAI_REQUEST_DATA)
+    return response.json()['data'][0]['embedding']
+
+
+class Cache(SQLModel, table=True):
+    __table_args__ = {
+        'mysql_TTL': f'created_at + INTERVAL {TIME_TO_LIVE} SECOND',
```
Comment (wd0517) [inline examples/semantic-cache/cache.py:50]: Provide a document link to introduce it.

File: examples/semantic-cache/cache.py (hunk 2 above)
Comment (sykp241095) [reply examples/semantic-cache/cache.py:11]: I removed all `async def` controller, so there is no need to use AsyncSession/AsyncEngine anymore.

File: examples/semantic-cache/cache.py (hunk 1 above)
Comment (sykp241095) [reply examples/semantic-cache/cache.py:126]: @wd0517 Any suggestions about the pre-filter or post-filter?

Comment (wd0517) [reply examples/semantic-cache/cache.py:126]: don't use filter in sql, filter in code

**Comment by sykp241095** [comment]
@wd0517 You should approve this before we can merge it as there is a branch protection policy...

**Review by wd0517 — APPROVED** [review]

**Pull request merged**